llms/mlx_lm/models/cache.py

```diff
@@ -320,6 +451,16 @@ def trim(self, n):
         self._idx -= n
         return n
 
+    def to_quantized(self, group_size: int = 64, bits: int = 4) -> QuantizedKVCache:
+        quant_cache = QuantizedKVCache(group_size=group_size, bits=bits)
+        quant_cache.offset = self.offset
+        if self.keys is not None:
+            quant_cache.keys = mx.quantize(self.keys, group_size=group_size, bits=bits)
+            quant_cache.values = mx.quantize(
+                self.values, group_size=group_size, bits=bits
+            )
+        return quant_cache
```

Comment:
I'm wondering if it makes sense to allow a rotating kv cache to be quantized? The quantized kv cache doesn't rotate as far as I can tell and if this cache is in a rotated state (e.g. the write head is on the same as the offset), then the behavior might is unclear / probably wrong?

Reply:
I think you're right, for now I've put in a NYI error.